core/templates/dev/head/dashboard/DashboardServices.js

```diff
@@ -0,0 +1,127 @@
+  NUM_VIEWS: 'num_views',
+  OPEN_FEEDBACK: 'num_open_threads',
+  UNRESOLVED_ANSWERS: 'num_unresolved_answers'
+});
+
+oppia.constant('HUMAN_READABLE_EXPLORATIONS_SORT_BY_KEYS', {
+  TITLE: 'Title',
+  RATING: 'Average Rating',
+  LAST_UPDATED: 'Last Updated',
+  NUM_VIEWS: 'Total Plays',
+  OPEN_FEEDBACK: 'New Feedback',
+  UNRESOLVED_ANSWERS: 'Unresolved Answers'
+});
+
+// Service for sorting the explorations based on different parameters.
+oppia.factory('sortExplorationsService', [
+  'utilsService', 'EXPLORATIONS_SORT_BY_KEYS', 'alertsService',
+  'RatingComputationService',
+  function(
+      utilsService, EXPLORATIONS_SORT_BY_KEYS, alertsService,
+      RatingComputationService) {
+    var TITLE_TEXT_DEFAULT = 'Untitled';
+    var AVERAGE_RATING_DEFAULT = 0.0;
+    var NUM_VALUE_DEFAULT = 0;
+
+    var sortByKey = function(explorationsList, key, isDescending) {
+      var result = angular.copy(explorationsList);
+      var valA, valB;
+      // JS Array.sort() method is stable in all major browsers except Chrome,
+      // since Chrome uses quicksort internally if the length of Array to be
+      // sorted is greater than 10. To make this stable, we maintain indexes of
+      // items in the array and return non-zero values from the function passed
+      // to .sort() method.
+      result = result.map(function(data, idx) {
+        return {
+          index: idx,
+          data: data
+        };
+      });
+      result.sort(function(a, b) {
+        var returnValue;
+        valA = a.data[key];
+        valB = b.data[key];
+        if (key === EXPLORATIONS_SORT_BY_KEYS.TITLE &&
+            utilsService.isString(valA)) {
+          if (valA === '') {
+            valA = TITLE_TEXT_DEFAULT;
+          }
+          if (valB === '') {
+            valB = TITLE_TEXT_DEFAULT;
+          }
+          valA = valA.toLowerCase();
+          valB = valB.toLowerCase();
+          returnValue = valA.localeCompare(valB);
+        } else if (key === EXPLORATIONS_SORT_BY_KEYS.RATING) {
+          valA = RatingComputationService.computeAverageRating(valA);
+          valB = RatingComputationService.computeAverageRating(valB);
+          if (!valA) {
+            valA = AVERAGE_RATING_DEFAULT;
+          }
+          if (!valB) {
+            valB = AVERAGE_RATING_DEFAULT;
+          }
+        } else {
+          if (typeof valA === 'undefined' || valA === null) {
+            valA = NUM_VALUE_DEFAULT;
+          }
+          if (typeof valB === 'undefined' || valB === null) {
+            valB = NUM_VALUE_DEFAULT;
+          }
+          returnValue = valA - valB;
+        }
+        // NOTE TO DEVELOPERS: Make sure the value returned here is non-zero to
+        // keep this sort stable.
+        return returnValue ? returnValue : (a.index - b.index);
+      });
+      result = result.map(function(value) {
+        return value.data;
+      });
+      if (isDescending || angular.equals(explorationsList, result)) {
```

Comment:
Had (incorrectly) thought of a fix for the list being reversed on every 2nd click (since the optional paramenter `isDescending` was being toggled each time and thus being passed as `true` as every 2nd time -- causing the original list to be reversed).
Changed the logic slightly. Now I simply change the sign of the return value inside `.sort(function() { ... })` if the order is descending!